src/System.ValueTuple/src/System/ValueTuple/ValueTuple.cs

```diff
@@ -255,7 +256,7 @@ internal static int CombineHashCodes(int h1, int h2, int h3)
 
         internal static int CombineHashCodes(int h1, int h2, int h3, int h4)
         {
-            return CombineHashCodes(CombineHashCodes(h1, h2), CombineHashCodes(h3, h4));
+            return CombineHashCodes(CombineHashCodes(h1, h2, h3), h4);
```

Comment:
@stephentoub This is actually just 3, since the 3-overload `CombineHashCodes` method (which will probably be inlined) is just calling the 2-overload one (which does the actual work) twice, and this calls the 2-overload one once. I mentioned why I changed it in my post: it 'spreads' the hash farther up for low numbers.

e.g. if you have a tuple of `1 2 3 4`, then in the previous implementation you would take the hash of 1 and 2, and since all the upper bits are 0 the furthest bit that would get affected is maybe the 6th bit. Then you hash 3 and 4 together, and the furthest bit that gets affected is the 7th bit. Then when you combine them again, the furthest bit that could be set is 11th, meaning you wasted ~21 of them.

With the new implementation it keeps spreading, so you would take 1 and 2, hash them to get 6 bits, hash it with 3 to get 11 bits, then with 4 to get 16 bits. Basically less bits are wasted, meaning a lower likelihood of collisions.